src/Filter/EntityFilter.php

```diff
@@ -0,0 +1,287 @@
+
+namespace App\Filter;
+
+use ApiPlatform\Api\IdentifiersExtractorInterface;
+use ApiPlatform\Api\IriConverterInterface;
+use ApiPlatform\Doctrine\Common\Filter\SearchFilterInterface;
+use ApiPlatform\Doctrine\Common\Filter\SearchFilterTrait;
+use ApiPlatform\Doctrine\Orm\Filter\AbstractFilter;
+use ApiPlatform\Doctrine\Orm\Util\QueryBuilderHelper;
+use ApiPlatform\Doctrine\Orm\Util\QueryNameGeneratorInterface;
+use ApiPlatform\Exception\InvalidArgumentException;
+use ApiPlatform\Metadata\Operation;
+use Doctrine\DBAL\Types\Types;
+use Doctrine\ORM\Query\Expr\Join;
+use Doctrine\ORM\QueryBuilder;
+use Doctrine\Persistence\ManagerRegistry;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\PropertyAccess\PropertyAccess;
+use Symfony\Component\PropertyAccess\PropertyAccessorInterface;
+use Symfony\Component\Serializer\NameConverter\NameConverterInterface;
+
+class EntityFilter extends AbstractFilter implements SearchFilterInterface
+{
+    use SearchFilterTrait;
+
+    public const DOCTRINE_INTEGER_TYPE = Types::INTEGER;
+
+    public const JSON_FILTER_NAME = 'filter';
+
+    public function __construct(
+        ManagerRegistry $managerRegistry,
+        IriConverterInterface $iriConverter,
+        PropertyAccessorInterface $propertyAccessor = null,
+        LoggerInterface $logger = null,
+        array $properties = null,
+        IdentifiersExtractorInterface $identifiersExtractor = null,
+        NameConverterInterface $nameConverter = null
+    ) {
+        parent::__construct($managerRegistry, $logger, $properties, $nameConverter);
+
+        $this->iriConverter = $iriConverter;
+        $this->identifiersExtractor = $identifiersExtractor;
+        $this->propertyAccessor = $propertyAccessor ?: PropertyAccess::createPropertyAccessor();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function apply(
+        QueryBuilder $queryBuilder,
+        QueryNameGeneratorInterface $queryNameGenerator,
+        string $resourceClass,
+        Operation $operation = null,
+        array $context = []
+    ): void {
+        if (in_array(self::JSON_FILTER_NAME, array_keys($context['filters']))) {
+            $jsonFilter = json_decode($context['filters'][self::JSON_FILTER_NAME], true);
+            unset($context['filters'][self::JSON_FILTER_NAME]);
+            $context['filters'] = array_merge($context['filters'], $jsonFilter);
+        }
+        foreach ($context['filters'] as $property => $value) {
+            $this->filterProperty($this->denormalizePropertyName($property), $value, $queryBuilder,
+                $queryNameGenerator, $resourceClass, $operation, $context);
+        }
+    }
+
+    protected function getIriConverter(): IriConverterInterface
+    {
+        return $this->iriConverter;
+    }
+
+    protected function getPropertyAccessor(): PropertyAccessorInterface
+    {
+        return $this->propertyAccessor;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function filterProperty(
```

Comment:
### 🤓 Nitpick: Something seems to be wrong with this func

Maybe it coz of me and it's ok for php, but it looks like very imperative-style hardcoded func. Can we have a call to try refactor it?)